src/google/protobuf/arenastring.h

```diff
@@ -200,8 +200,12 @@ class PROTOBUF_EXPORT TaggedStringPtr {
   void* ptr_;
 };
 
-static_assert(std::is_trivial<TaggedStringPtr>::value,
-              "TaggedStringPtr must be trivial");
+static_assert(std::is_trivially_default_constructible<TaggedStringPtr>::value,
```

Comment:
I added these 3 checks, however I am unsure if we also want to have any check about being trivially copyable. I see the class implements some specific `Copy` methods, so I guess we are not meant to copy it via any copy operator/constructor, that's why I omitted the trivial copy check.